### PR TITLE
[BLCKCHN-209] Add Defi Module to py client

### PR DESCRIPTION
**Pull request by yeehan-crypto-com**
# Pull Request Template

## Description
Module usage
```
from crypto_com_developer_platform_client import (
    Defi,
    DefiProtocol,
)
from crypto_com_developer_platform_client.interfaces.defi_interfaces import DefiProtocol


tokens = Defi.get_whitelisted_tokens(DefiProtocol.VVS)
farms = Defi.get_all_farms(DefiProtocol.H2)
farm = Defi.get_farm_by_symbol(DefiProtocol.H2, "zkCRO-MOON")
```

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:
Before you submit the pull request, please review the following checklist:
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

## Linked Issues
Mention the issues that are addressed by this PR. You can use the `#` symbol to link an issue.

### CODE REVIEW

File: crypto_com_developer_platform_client/integrations/defi_api.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+def get_whitelisted_tokens(project: str, api_key: str) -> ApiResponse:
+    """
+    Get whitelisted tokens for a specific DeFi project.
+
+    :param project: The DeFi project name
+    :param api_key: The API key for authentication
+    :return: List of whitelisted tokens
+    :raises Exception: If the request fails or server responds with an error
+    """
+    url = f"""http://localhost:8001/v1/cdc-developer-platform/defi/whitelisted-tokens/{
```
Comment (rarcifa) [inline crypto_com_developer_platform_client/integrations/defi_api.py:17]: http://localhost:8001? Should this be the prod url?

File: crypto_com_developer_platform_client/integrations/defi_api.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+def get_all_farms(project: str, api_key: str) -> ApiResponse:
+    """
+    Get all farms for a specific DeFi project.
+
+    :param project: The DeFi project name
+    :param api_key: The API key for authentication
+    :return: List of all farms
+    :raises Exception: If the request fails or server responds with an error
+    """
+    url = f"""http://localhost:8001/v1/cdc-developer-platform/defi/farms/{
```
Comment (rarcifa) [inline crypto_com_developer_platform_client/integrations/defi_api.py:44]: and here

File: crypto_com_developer_platform_client/integrations/defi_api.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    """
+    Get specific farm information by symbol for a DeFi project.
+
+    :param project: The DeFi project name
+    :param symbol: The farm symbol
+    :param api_key: The API key for authentication
+    :return: Information about the specific farm
+    :raises Exception: If the request fails or server responds with an error
+    """
+    url = f"""http://localhost:8001/v1/cdc-developer-platform/defi/farms/{
```
Comment (rarcifa) [inline crypto_com_developer_platform_client/integrations/defi_api.py:72]: and here

**Comment by yeehan-crypto-com** [comment]
> looks good but check the urls

Yeah i think i should abstract them out to a base URL to prevent future issues like this, thanks for the heads up

**Review by rarcifa — APPROVED** [review]

**Pull request merged**